tests/models/test_default_evaluator.py

```diff
@@ -499,3 +502,160 @@ def test_gen_multiclass_roc_curve():
 
     expected_auc = [0.75, 0.75, 0.3333]
     assert np.allclose(results.auc, expected_auc, rtol=1e-3)
+
+
+def test_evaluate_custom_metric_incorrect_return_formats():
+    eval_df = pd.DataFrame({"prediction": [1.2, 1.9, 3.2], "target": [1, 2, 3]})
+    metrics = _get_regressor_metrics(eval_df["target"], eval_df["prediction"])
+
+    def dummy_fn(*_):
```

Comment:
Also add a test on lambda function. e.g. legal case: `custom_metrics=(lambda x: {'square': x**2},)` and illegal case: `custom_metrics=(lambda x: x,)`